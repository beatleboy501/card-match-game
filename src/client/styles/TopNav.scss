.appBar {
  display: flex !important;
  flex-direction: row !important;
  width: 80% !important;
  margin-top: 1em !important;
  padding: 0 55px 0 5px !important;

  button {
    align-self: center !important;
  }

  section {
    position: absolute;
    right: 15%;
    align-self: center;
  }
}

.header {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  padding: 10px 0;

  img {
    margin: 10px;
    outline: 0.5px solid #000000;
  }
}

.toolbar {
  width: 100%;
  justify-content: center;
}
