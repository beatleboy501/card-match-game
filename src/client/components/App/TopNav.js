import React from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import Timer from '../MemoryGame/Timer';
import styles from '../../styles/TopNav.scss';

const propTypes = {
  currentCards: PropTypes.arrayOf(PropTypes.object),
  logo: PropTypes.string,
  onBurgerClick: PropTypes.func,
  timerRef: PropTypes.func,
};

const defaultProps = {
  currentCards: null,
  logo: '',
  onBurgerClick: () => {},
  timerRef: () => {},
};

const TopNav = (props) => {
  const {
    currentCards, logo, onBurgerClick, timerRef,
  } = props;
  return (
    <AppBar
      position="static"
      color="default"
      className={styles.appBar}
      elevation={10}
    >
      <IconButton
        color="inherit"
        aria-label="Open drawer"
        onClick={onBurgerClick}
        className={styles.burger}
      >
        <MenuIcon />
      </IconButton>
      <Toolbar className={styles.toolbar}>
        <div className={styles.header}>
          <img src={logo} alt="logo" className={styles.logo} />
          <Typography variant="title" color="inherit">
            Card Match Game
          </Typography>
        </div>
      </Toolbar>
      {currentCards && <Timer ref={el => timerRef(el)} />}
    </AppBar>
  );
};

TopNav.propTypes = propTypes;
TopNav.defaultProps = defaultProps;

export default TopNav;
