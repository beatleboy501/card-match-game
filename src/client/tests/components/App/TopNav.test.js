import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TopNav from '../../../components/App/TopNav';
import { easyCards } from '../../testUtils';

Enzyme.configure({ adapter: new Adapter() });

const props = {
  currentCards: easyCards,
  logo: 'logo',
  parentRef: jest.fn(() => {}),
};

let component;

it('Renders without crashing', () => {
  component = shallow(<TopNav props={props} />).shallow();
  expect(component).toMatchSnapshot();
});
