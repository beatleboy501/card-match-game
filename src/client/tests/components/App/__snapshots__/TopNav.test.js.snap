// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(AppBar)
    className="appBar"
    color="default"
    elevation={10}
    position="static"
  >
    <WithStyles(IconButton)
      aria-label="Open drawer"
      className="burger"
      color="inherit"
      onClick={[Function]}
    >
      <pure(MenuIcon) />
    </WithStyles(IconButton)>
    <WithStyles(Toolbar)
      className="toolbar"
    >
      <div
        className="header"
      >
        <img
          alt="logo"
          className="logo"
          src=""
        />
        <WithStyles(Typography)
          color="inherit"
          variant="title"
        >
          Card Match Game
        </WithStyles(Typography)>
      </div>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <WithStyles(IconButton)
          aria-label="Open drawer"
          className="burger"
          color="inherit"
          onClick={[Function]}
        >
          <pure(MenuIcon) />
        </WithStyles(IconButton)>,
        <WithStyles(Toolbar)
          className="toolbar"
        >
          <div
            className="header"
          >
            <img
              alt="logo"
              className="logo"
              src=""
            />
            <WithStyles(Typography)
              color="inherit"
              variant="title"
            >
              Card Match Game
            </WithStyles(Typography)>
          </div>
        </WithStyles(Toolbar)>,
        null,
      ],
      "className": "appBar",
      "classes": Object {
        "colorDefault": "MuiAppBar-colorDefault-7",
        "colorPrimary": "MuiAppBar-colorPrimary-8",
        "colorSecondary": "MuiAppBar-colorSecondary-9",
        "positionAbsolute": "MuiAppBar-positionAbsolute-3",
        "positionFixed": "MuiAppBar-positionFixed-2",
        "positionRelative": "MuiAppBar-positionRelative-6",
        "positionStatic": "MuiAppBar-positionStatic-5",
        "positionSticky": "MuiAppBar-positionSticky-4",
        "root": "MuiAppBar-root-1",
      },
      "color": "default",
      "elevation": 10,
      "position": "static",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "aria-label": "Open drawer",
          "children": <pure(MenuIcon) />,
          "className": "burger",
          "color": "inherit",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="header"
          >
            <img
              alt="logo"
              className="logo"
              src=""
            />
            <WithStyles(Typography)
              color="inherit"
              variant="title"
            >
              Card Match Game
            </WithStyles(Typography)>
          </div>,
          "className": "toolbar",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="logo"
                src=""
              />,
              <WithStyles(Typography)
                color="inherit"
                variant="title"
              >
                Card Match Game
              </WithStyles(Typography)>,
            ],
            "className": "header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "logo",
                "src": "",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Card Match Game",
                "color": "inherit",
                "variant": "title",
              },
              "ref": null,
              "rendered": "Card Match Game",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      null,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <WithStyles(IconButton)
            aria-label="Open drawer"
            className="burger"
            color="inherit"
            onClick={[Function]}
          >
            <pure(MenuIcon) />
          </WithStyles(IconButton)>,
          <WithStyles(Toolbar)
            className="toolbar"
          >
            <div
              className="header"
            >
              <img
                alt="logo"
                className="logo"
                src=""
              />
              <WithStyles(Typography)
                color="inherit"
                variant="title"
              >
                Card Match Game
              </WithStyles(Typography)>
            </div>
          </WithStyles(Toolbar)>,
          null,
        ],
        "className": "appBar",
        "classes": Object {
          "colorDefault": "MuiAppBar-colorDefault-7",
          "colorPrimary": "MuiAppBar-colorPrimary-8",
          "colorSecondary": "MuiAppBar-colorSecondary-9",
          "positionAbsolute": "MuiAppBar-positionAbsolute-3",
          "positionFixed": "MuiAppBar-positionFixed-2",
          "positionRelative": "MuiAppBar-positionRelative-6",
          "positionStatic": "MuiAppBar-positionStatic-5",
          "positionSticky": "MuiAppBar-positionSticky-4",
          "root": "MuiAppBar-root-1",
        },
        "color": "default",
        "elevation": 10,
        "position": "static",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "aria-label": "Open drawer",
            "children": <pure(MenuIcon) />,
            "className": "burger",
            "color": "inherit",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="header"
            >
              <img
                alt="logo"
                className="logo"
                src=""
              />
              <WithStyles(Typography)
                color="inherit"
                variant="title"
              >
                Card Match Game
              </WithStyles(Typography)>
            </div>,
            "className": "toolbar",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="logo"
                  className="logo"
                  src=""
                />,
                <WithStyles(Typography)
                  color="inherit"
                  variant="title"
                >
                  Card Match Game
                </WithStyles(Typography)>,
              ],
              "className": "header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "logo",
                  "className": "logo",
                  "src": "",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Card Match Game",
                  "color": "inherit",
                  "variant": "title",
                },
                "ref": null,
                "rendered": "Card Match Game",
                "type": [Function],
              },
            ],
            "type": "div",
          },
          "type": [Function],
        },
        null,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
